backend/services/AccountService.ts

```diff
@@ -0,0 +1,217 @@
+import * as Config from '../config/config';
+import { HttpStatusCode } from 'axios';
+import { HttpError } from '../types/HttpError';
+import { Request } from 'express';
+import logger from '../logger';
+import { WebhookController } from '../controllers/WebhookController';
+
+type AccountResponse = Components.Schemas.AccountResponse;
+type WalletLoginRequestDto = Components.Schemas.WalletLoginRequestDto;
+type WalletLoginConfigResponse = Components.Schemas.WalletLoginConfigResponse;
+type WalletLoginResponse = Components.Schemas.WalletLoginResponse;
+
+/**
+ * The `AccountService` class provides methods for interacting with user accounts.
+ * It handles operations such as retrieving account information, signing in or signing up users,
+ * and managing authentication tokens. It uses the Account-Service RESTful API to perform these operations.
+ */
+export class AccountService {
+  private static instance: AccountService;
+  private _client: AccountServiceClient;
+
+  private constructor() {}
+
+  public static async getInstance(): Promise<AccountService> {
+    if (!AccountService.instance) {
+      AccountService.instance = new AccountService();
+      await AccountService.instance.connect();
+    }
+    return AccountService.instance;
+  }
+
+  private async connect() {
+    if (this._client === undefined) {
+      const api = new OpenAPIClientAxios({
+        definition: openapiJson as Document,
+        withServer: { url: Config.instance().accountServiceUrl },
+      });
+      this._client = await api.init<AccountServiceClient>();
+    }
+  }
+
+  private set client(api: AccountServiceClient) {
+    this._client = api;
+  }
+
+  private get client() {
+    if (this._client === undefined) {
+      throw new Error(`${this.constructor.name} API not initialized`);
+    }
+    return this._client;
+  }
+
+  /**
+   * Retrieves the SIWF configuration for wallet login.
+   * @returns A promise that resolves to a `WalletLoginConfigResponse` object.
+   * @throws If there was an error retrieving the SWIF config.
+   */
+  public async getSWIFConfig(): Promise<WalletLoginConfigResponse> {
+    try {
+      const response = await this.client.AccountsController_getSIWFConfig();
+      return response.data;
+    } catch (e) {
+      logger.error('Failed to get SIWF config: ', e);
+      throw e;
+    }
+  }
+
+  /**
+   * Retrieves the account information for a given MSA ID.
+   * @param msaId - The MSA ID of the account to retrieve.
+   * @returns A Promise that resolves to an AccountResponse object containing the account information.
+   * @throws If there was an error retrieving the account information.
+   */
+  public async getAccount(msaId: string): Promise<AccountResponse> {
+    try {
+      const response = await this.client.AccountsController_getAccount(msaId);
+      logger.debug(`AccountService: getAccount: Got account for msaID:(${msaId}), data:(${JSON.stringify(response.data)})`);
+      return {
+        msaId: parseInt(msaId),
```

Comment:
Follow up: creating an issue to update msaId to be the same type everywhere, currently some are string and some are number.

Reply:
👍🏻 
correct type should be 'string' (or bigint, but we use string most other places...)